llvm/lib/Target/AMDGPU/SILoadStoreOptimizer.cpp

```diff
@@ -2571,3 +2580,19 @@ bool SILoadStoreOptimizer::runOnMachineFunction(MachineFunction &MF) {
 
   return Modified;
 }
+
+PreservedAnalyses
+SILoadStoreOptimizerPass::run(MachineFunction &MF,
+                              MachineFunctionAnalysisManager &MFAM) {
```

Comment:
Missing the code that skips this optimization for -O0 (optNone). Insert the following code here.
if (MF.getFunction().hasOptNone())
   return false;